solr/core/src/java/org/apache/solr/handler/admin/CollectionsHandler.java

```diff
@@ -1791,14 +1794,17 @@ public Map<String, Object> execute(
     REPLACENODE_OP(
         REPLACENODE,
         (req, rsp, h) -> {
-          return copy(
-              req.getParams(),
-              null,
-              "source", // legacy
-              "target", // legacy
-              WAIT_FOR_FINAL_STATE,
-              CollectionParams.SOURCE_NODE,
-              CollectionParams.TARGET_NODE);
+          final SolrParams params = req.getParams();
+          final RequiredSolrParams requiredParams = req.getParams().required();
+          final ReplaceNodeAPI.ReplaceNodeRequestBody requestBody =
+              new ReplaceNodeAPI.ReplaceNodeRequestBody();
+          requestBody.targetNodeName = params.get(TARGET_NODE);
+          requestBody.waitForFinalState = params.getBool(WAIT_FOR_FINAL_STATE, false);
```

Comment:
[0] Prior to this PR, if the user didn't explicitly specify a waitForFinalState value, the created "message" wouldn't contain that key at all.

The use of `getBool(key, defaultVal)` here means that the constructed RNRB will _always_ have a waitForFinalState value that gets added to the "message" that the v2 code builds.  This isn't a problem necessarily; the docs say that this parameter defaults to "false" - so this shouldn't change any behavior.  But it is something to keep an eye on if test failures crop up later.

If you wanted to be more conservative though, you could drop the default-val here to keep closer in line with the existing behavior.